examples/oidc/css-api/nextjs/src/hooks/useGetData.js

```diff
@@ -0,0 +1,33 @@
+import { useState } from 'react'
+
+/**
+ * Hook to manage loading and error state on API get requests.
+ * 
+ * @param {*} defaultData Data to use before fetch is complete 
+ * @returns {Object} {data, loadingData, apiError, fetchData}
+ */
+export default function useGetData(defaultData) {
+    const [data, setData] = useState(defaultData);
+    const [loadingData, setLoadingData] = useState(false);
+    const [apiError, setApiError] = useState(false);
+
+    const fetchData = async (url) => {
+        try {
+            setData(defaultData);
+            setApiError(false);
+            setLoadingData(true);
+            const fetchedData = await fetch(url)
```

Comment:
@jlangy the logic does not handle when there is error and data is null in the response. I get showed `no integrations found` error in this scenario

Reply:
I see what you mean, I added in some more robust handling for fetching in the API side so the status code propogates to the user